images/virtualization-artifact/pkg/controller/service/ip_address_service.go

```diff
@@ -53,73 +54,72 @@ func NewIpAddressService(
 	}
 }
 
-func (s IpAddressService) IsAvailableAddress(address string, allocatedIPs common.AllocatedIPs) bool {
-	ip := net.ParseIP(address)
+func (s IpAddressService) IsAvailableAddress(address string, allocatedIPs common.AllocatedIPs) error {
+	ip, err := netip.ParseAddr(address)
+	if err != nil || !ip.IsValid() {
+		return ErrInvalidIpAddress
```

Comment:
Do we really need var for this err?

Reply:
Sorry, I don't quite understand what the problem is.

Reply:
Sorted it out. Fixed.